lib/containers/container_images.pm

```diff
@@ -156,20 +155,24 @@ sub test_opensuse_based_image {
 }
 
 sub ensure_container_rpm_updates {
-    my $diff_file   = shift;
-    my $regex2match = qr/^-(?<package>[^\s]+)\s+(\d+\.?\d+?)+-(?P<update_version>\d+\.?\d+?\.\d+\b).*\s+(\d+\.?\d+?)+-(?P<stable_version>\d+\.?\d+?\.\d+\b)/;
-    my $context     = script_output "cat $diff_file";
+    my $diff_file     = shift;
+    my $regex2match   = qr/^-(?<package>[^\s]+)\s+(\d+\.?\d+?)+-(?P<update_version>\d+\.?\d+?\.\d+\b).*\s+(\d+\.?\d+?)+-(?P<stable_version>\d+\.?\d+?\.\d+\b)/;
+    my $regex2zerorpm = qr/^Version differences: None$/;
+    my $context       = script_output "cat $diff_file";
     open(my $data, '<', \$context) or die "problem with $diff_file argument", $!;
     while (my $line = <$data>) {
+        if ($line =~ $regex2zerorpm) {
+            record_info("No Update found", "no updates found between rpm versions");
+            last;
+        }
         if ($line =~ $regex2match) {
             # Use of Dotted-Decimal-Versions. Do not remove 'v' prefix
             my $updated_v = version->parse("v$+{update_version}");
             my $stable_v  = version->parse("v$+{stable_version}");
             record_info("checking... $+{package}", "$+{package} $stable_v to $updated_v");
-            assert_true($updated_v > $stable_v, "$+{package} $stable_v is not updated to $updated_v");
+            die "$+{package} $stable_v is not updated to $updated_v" unless ($updated_v > $stable_v);
         }
```

Comment:
What about `else { … }` statement? It might be useful for incidental debugging.

Reply:
else what... not sure. i assume that you talk about the $line matching. thus if the regex is correct will match only the ones that have the packages. i do not think we care for all the other $lines atm.

Reply:
I think Pavel means the following:

        if ($line =~ $regex2zerorpm) {
        ...
        } elsif ($line =~ $regex2match) {
        ...
        } else {
            // we don't care about the other lines for now
        }

And to add a check, if at least one of the checks is happening.

Reply:
xm. i believe it is the same. but now you make think which version is more computational efficient. let me try.

Reply:
if..else seems faster than multi_ifs. and counting also the prediction factor i ended up with switching the blocks. i hope it looks better now(reopen it if you feel that something is still missing). thanks for the inspiration.